extensions/wikia/SitemapXml/SpecialSitemapXmlController.class.php

```diff
@@ -73,60 +143,99 @@ private function getIso8601Timestamp( $touched ) {
 			$touched[12] . $touched[13] . 'Z';
 	}
 
-	private function getNamespaces() {
+	private function getOtherNamespaces() {
 		$nsIds = array_unique( $this->wg->ContLang->getNamespaceIds() );
+		return array_filter( $nsIds, [ $this, 'isOtherNamespace' ] );
+	}
 
-		return array_filter( $nsIds, function ( $id ) {
-			// Exclude negative namespaces
-			if ( $id < 0 ) {
-				return false;
-			}
+	private function isOtherNamespace( $nsId ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)